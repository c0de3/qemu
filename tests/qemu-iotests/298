#!/usr/bin/env python3
#
# Test for preallocate filter
#
# Copyright (c) 2020 Virtuozzo International GmbH.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

import os
import iotests
from iotests import log

iotests.script_initialize(supported_fmts=['qcow2'])

size = 10 * 1024 * 1024
disk = iotests.file_path('disk')

iotests.qemu_img_create('-f', iotests.imgfmt, disk, str(size))

opts = f'driver={iotests.imgfmt},' \
    f'file.driver=preallocate,file.file.filename={disk}'
p = iotests.QemuIoInteractive('--image-opts', '-t', 'none', opts)

log(p.cmd('write 0 1M'), filters=[iotests.filter_qemu_io])
p.cmd('flush')

if os.path.getsize(disk) > 100 * 1024 * 1024:
    log('file in progress is big, preallocation works')

p.close()

if os.path.getsize(disk) < 10 * 1024 * 1024:
    log('file is small after close')
