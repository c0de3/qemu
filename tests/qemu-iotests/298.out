wrote 1048576/1048576 bytes at offset 0
1 MiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

file in progress is big, preallocation works
file is small after close
